lib/Db/Feed.php

```diff
@@ -326,15 +326,17 @@ public function jsonSerialize(): array
             'basicAuthPassword'
         ]);
 
-        $url = parse_url($this->link, PHP_URL_HOST);
-
-        // strip leading www. to avoid css class confusion
-        if (strpos($url, 'www.') === 0) {
-            $url = substr($url, 4);
+        if (!is_null($this->link)) {
+            $url = parse_url($this->link, PHP_URL_HOST);
+
+            // strip leading www. to avoid css class confusion
+            if (strpos($url, 'www.') === 0) {
+                $url = substr($url, 4);
+            }
+    
+            $serialized['cssClass'] = 'custom-' . str_replace('.', '-', $url);
         }
```

Comment:
```suggestion
        if (is_null($this->link)) {
          return $serialized;
        }

        $url = parse_url($this->link, PHP_URL_HOST);

        // strip leading www. to avoid css class confusion
        if (strpos($url, 'www.') === 0) {
            $url = substr($url, 4);
        }
    
        $serialized['cssClass'] = 'custom-' . str_replace('.', '-', $url);
```